client/my-sites/stats/activity-log/unavailability-notice.jsx

```diff
@@ -0,0 +1,60 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Banner from 'components/banner';
+import { getSiteAdminUrl } from 'state/sites/selectors';
+import { getSelectedSiteSlug } from 'state/ui/selectors';
+import { getRewindState } from 'state/selectors';
+
+export const UnavailabilityNotice = ( { adminUrl, reason, rewindState, slug, translate } ) => {
+	if ( rewindState !== 'unavailable' ) {
+		return null;
+	}
+
+	switch ( reason ) {
+		case 'no_connected_jetpack':
+			return (
+				<Banner
+					icon="history"
+					href={ adminUrl }
+					title={ translate( 'The site cannot be accessed' ) }
+					description={ translate(
+						'We are unable to create backups and Rewind because we cannot access your site.'
+					) }
+				/>
+			);
+
+		case 'vp_can_transfer':
+			return (
+				<Banner
+					icon="history"
+					href={ `/settings/security/${ slug }` }
+					title={ translate( 'VaultPress is running' ) }
+					description={ translate(
+						'We are unable to create backups and Rewind while VaultPress is running.'
+					) }
+				/>
+			);
+	}
+};
+
+const mapStateToProps = ( state, { siteId } ) => {
+	const { reason, state: rewindState } = getRewindState( state, siteId );
+
+	return {
+		adminUrl: getSiteAdminUrl( state, siteId ),
+		reason,
+		rewindState,
+		slug: getSelectedSiteSlug( state ),
+	};
+};
+
+export default connect( mapStateToProps )( localize( UnavailabilityNotice ) );
```

Comment:
In most places we (even you) placed the contents of `mapStateToProps` inline in the connect without collecting it first in a constant. Why this pattern now?

Reply:
Whether I've inlined it I've always preferred explicitly separating them for consistency and explicitness. I've not made a big deal about it because we share a variety of preferences and it doesn't have a big impact. Also, inlining `mapDispatchToProps` when we have no arguments seems much more tolerable than `mapStateToProps` which _always_ needs args